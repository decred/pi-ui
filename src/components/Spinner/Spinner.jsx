import React from "react";
import styles from "./styles.module.css";
import PropTypes from "prop-types";

const Spinner = ({ invert, width, height }) => {
  return (
    <div
      className={invert ? styles.spinnerInvert : styles.spinner}
      style={{ width, height }}
    />
  );
};

Spinner.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
  invert: PropTypes.bool,
};

export default Spinner;
