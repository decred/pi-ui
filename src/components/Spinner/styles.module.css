.spinner {
  display: inline-block;
  width: 1.8rem;
  height: 1.8rem;
  border: 0.2rem solid rgba(255, 255, 255, 0.3);
  border-radius: 50%;
  border-top-color: var(--color-white);
  animation: spin 1s ease-in-out infinite;
  -webkit-animation: spin 1s ease-in-out infinite;
}

.spinnerInvert {
  composes: spinner;
  border: 0.2rem solid rgba(9, 20, 64, 0.3);
  border-top-color: var(--color-primary-dark);
}

@keyframes spin {
  to {
    -webkit-transform: rotate(360deg);
  }
}
@-webkit-keyframes spin {
  to {
    -webkit-transform: rotate(360deg);
  }
}
