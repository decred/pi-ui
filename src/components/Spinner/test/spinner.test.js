import React from "react";
import Spinner from "../Spinner";
import { create } from "react-test-renderer";

describe("Spinner component", () => {
  test("Matches the snapshot", () => {
    const spinner = create(<Spinner />);
    expect(spinner.toJSON()).toMatchSnapshot();
  });
});
