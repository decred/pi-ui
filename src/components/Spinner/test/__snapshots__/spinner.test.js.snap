// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component Matches the snapshot 1`] = `
<div
  className="spinner"
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
/>
`;
